ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs

```diff
@@ -508,32 +534,36 @@ data TraceEvent blk
     -- ^ An old or invalid on-disk snapshot was deleted
   deriving (Generic, Eq, Show)
 
+
+-- | Which point the replay started from
+newtype ReplayStart blk = ReplayStart (Point blk) deriving (Eq, Show)
+
+-- | Which point the replay is expected to end at
+newtype ReplayGoal blk = ReplayGoal (Point blk) deriving (Eq, Show)
+
 -- | Events traced while replaying blocks against the ledger to bring it up to
 -- date w.r.t. the tip of the ImmutableDB during initialisation. As this
 -- process takes a while, we trace events to inform higher layers of our
 -- progress.
---
--- The @replayTo@ parameter is meant to be filled in by a higher layer,
--- i.e., the ChainDB.
-data TraceReplayEvent blk replayTo
-  = ReplayFromGenesis replayTo
+data TraceReplayEvent blk
+  = ReplayFromGenesis (ReplayGoal blk)
     -- ^ There were no LedgerDB snapshots on disk, so we're replaying all
     -- blocks starting from Genesis against the initial ledger.
     --
     -- The @replayTo@ parameter corresponds to the block at the tip of the
     -- ImmutableDB, i.e., the last block to replay.
-  | ReplayFromSnapshot DiskSnapshot (RealPoint blk) replayTo
+  | ReplayFromSnapshot DiskSnapshot (RealPoint blk) (ReplayStart blk) (ReplayGoal blk)
     -- ^ There was a LedgerDB snapshot on disk corresponding to the given tip.
     -- We're replaying more recent blocks against it.
     --
     -- The @replayTo@ parameter corresponds to the block at the tip of the
     -- ImmutableDB, i.e., the last block to replay.
-  | ReplayedBlock (RealPoint blk) [LedgerEvent blk] replayTo
+  | ReplayedBlock (RealPoint blk) [LedgerEvent blk] (ReplayStart blk) (ReplayGoal blk)
     -- ^ We replayed the given block (reference) on the genesis snapshot
     -- during the initialisation of the LedgerDB.
     --
     -- The @blockInfo@ parameter corresponds replayed block and the @replayTo@
     -- parameter corresponds to the block at the tip of the ImmutableDB, i.e.,
     -- the last block to replay.
   | UpdateLedgerDbTraceEvent (UpdateLedgerDbTraceEvent blk)
-  deriving (Generic, Eq, Show, Functor, Foldable, Traversable)
+  deriving (Generic, Eq, Show)
```

Comment:
Minor: Can you also remove the `DeriveFunctor`, `DeriveFoldable`, and `DeriveTraversable` LANGUAGE pragmas now?

Reply:
Addressed